vespa_analysis/vespa_pair.py

```diff
@@ -0,0 +1,79 @@
+import os
+import argparse
+from Bio import SeqIO
+import pandas as pd
+
+def gen_score(ref_seq,gen_seq,ref_aln,gen_aln,vesp_dict):
+    scores = dict()
+    c = 0
+    gc = 0
+    g_inx =[]
+    g_start = 0
+    if gen_aln == ref_aln:
+        scores['NA'] = 0
+        g_inx.append(0)
+    else:
+        if len(gen_seq) != len(gen_aln.replace('-','')):
+            g_inx_ter = gen_seq.replace(gen_aln.replace('-',''),'-').split('-')
+            g_start = len(g_inx_ter[0])
+        if len(ref_seq) != len(ref_aln.replace('-','')):
+            ref_ter_list = ref_seq.replace(ref_aln.replace('-',''),'-').split('-')
+            gen_ter_list = ['-'*len(ref_ter_list[0]), '-'*len(ref_ter_list[1])]
+            ref_aln = ref_ter_list[0]+ref_aln+ref_ter_list[1]
+            gen_aln = gen_ter_list[0]+gen_aln+gen_ter_list[1]
+
+        for j in range(len(ref_aln)):
+            if gen_aln[j]!='-':
+                gc = gc+1
+            if ref_aln[j] != '-':
+                c = c+1
+                if (gen_aln[j] != '-') and (ref_aln[j] != gen_aln[j]):
+                    scores[ref_aln[j]+str(c-1)+gen_aln[j]] = vesp_dict[ref_aln[j]+str(c-1)+gen_aln[j]]
+                    g_inx.append(gc-1+g_start)
+    return [scores,ref_aln,gen_aln,g_inx]
+def arg_parser():
+    parser = argparse.ArgumentParser(description='Maps VESPA scores to mutant sequences')
+    parser.add_argument('--mmseq_out', type=str, action='store', help='FULL Path of mmseqs alignment output',required=True)
+    parser.add_argument('--vespa_out', type=str, action='store', help='FULL Path of vespa output csv file',required=True)
+    parser.add_argument('--ref_id', type=str, action='store', help='reference sequence ID',required=True)
+    parser.add_argument('--ref_seq', type=str, action='store', help='reference sequence',required=True)
+    parser.add_argument('--gen_id', type=str, action='store', help='generated sequence ID',required=True)
+    parser.add_argument('--gen_seq', type=str, action='store', help='generated sequence',required=True)
+    args = parser.parse_args()
+
+    mmseq_out = args.mmseq_out
+    vesp_out = args.vespa_out
+    ref_id = args.ref_id
+    ref_seq = args.ref_seq
+    gen_id = args.gen_id
+    gen_seq = args.gen_seq
+    return [mmseq_out,vesp_out,ref_id,ref_seq,gen_id,gen_seq]
+
+##Parsing Inputs
```

Comment:
you can add this condition: if __name__ == '__main__' and you can add all of the following lines starting from line 53 to line 79 under this condition.